src/SFA.DAS.FAT.Web/Controllers/CoursesController.cs

```diff
@@ -183,6 +189,18 @@ public async Task<IActionResult> CourseProviders(GetCourseProvidersRequest reque
                     return RedirectToRoute(RouteNames.CourseDetails, new { request.Id });
                 }
 
+                courseProvidersViewModel.HomeUrl = Url.RouteUrl(RouteNames.ServiceStart)!;
+                courseProvidersViewModel.ShortListUrl = Url.RouteUrl(RouteNames.ShortList)!;
+                courseProvidersViewModel.ShortListItemCount = result.ShortlistItemCount;
+
+                courseProvidersViewModel.ApprenticeTrainingCoursesUrl = location?.Name == null ? Url.RouteUrl(RouteNames.Courses) : Url.RouteUrl(RouteNames.Courses, new { Location = location.Name })!;
+
+                courseProvidersViewModel.ApprenticeTrainingCourseUrl = location?.Name == null ?
+                    Url.RouteUrl(RouteNames.CourseDetails, new { request.Id }) :
+                    Url.RouteUrl(RouteNames.CourseDetails, new { request.Id, Location = location.Name })!;
+
+                courseProvidersViewModel.ApprenticeCourseTitle = courseProvidersViewModel.Course.TitleAndLevel;

```

Comment:
The URLs dont change, so there is possibility that we can calculate the URLs in one place and then set Show flags according to the need of the page.

Reply:
we could indeed do that, in which case we'd just use the booleans and set them to true in the viewmodels.

Reply:
this is now all reworked to use just booleans, and the pageLinks basically contains all it needs to work given the booleans are set or not